runtime/src/main/java/org/corfudb/runtime/object/transactions/SnapshotTransactionalContext.java

```diff
@@ -85,4 +85,9 @@ public SnapshotTransactionalContext(Transaction transaction) {
     public void addTransaction(AbstractTransactionalContext tc) {
         throw new UnsupportedOperationException("Can't merge into a readonly txn (yet)");
     }
+
+    @Override
+    public void addPreCommitListener(TransactionalContext.PreCommitListener preCommitListener) {
+        throw new UnsupportedOperationException("Can't register precommit hooks in readonly txn");
```

Comment:
are snapshot transactions considered read-only?

Reply:
yes, there are similar exceptions for attempting to write in a snapshot transaction